eng/source-build/build.sh

```diff
@@ -1,5 +1,8 @@
 #!/usr/bin/env bash
 
+echo "git config --global protocol.file.allow always"
```

Comment:
Can we do it explicitly for the submodule command instead of setting it globally?

I'm ok either way for now.

Reply:
Created a follow-up issue to address this comment. https://github.com/NuGet/Client.Engineering/issues/1928. Hope you are okay with merging this right now to unblock builds.

Reply:
Yep, see teams channel reply as well.